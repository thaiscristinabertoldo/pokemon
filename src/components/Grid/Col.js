import styled from 'styled-components'

export const Col = styled.div`
  flex: ${(prop) => prop.size};
  margin: ${(prop) => prop.margin};
  width: ${(prop) => prop.width};
`

Col.defaultProps = {
  size: 1,
  margin: '10px 20px',
  width: 'inherit',
}
