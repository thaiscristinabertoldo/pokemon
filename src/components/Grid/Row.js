import styled from 'styled-components'

export const Row = styled.div`
  display: flex;
  flex-direction: ${(prop) => prop.flexDirection};
  justify-content: ${(prop) => prop.justifyContent};
  align-items: ${(prop) => prop.alignItems};
  width: ${(prop) => prop.width};
`

Row.defaultProps = {
  flexDirection: 'row',
  justifyContent: 'flex-start',
  alignItems: 'flex-start',
  width: 'unset',
}
