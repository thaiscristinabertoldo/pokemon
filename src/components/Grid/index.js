export * from './Grid'
export * from './Row'
export * from './Col'
