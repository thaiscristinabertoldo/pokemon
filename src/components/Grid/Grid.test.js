import React from 'react'

import { render, cleanup } from '@testing-library/react'

import { Grid } from './Grid'

afterEach(cleanup)

describe('Grid', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Grid />)

    expect(asFragment(<Grid />)).toMatchSnapshot()
  })
})
