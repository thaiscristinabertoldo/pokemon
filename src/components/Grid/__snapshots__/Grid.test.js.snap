// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm StDqN"
  />
</DocumentFragment>
`;
